app/validation/validators.py

```
@@ -0,0 +1,129 @@
+from datetime import datetime
+
+from app.validation.error_messages import error_messages
+
+from wtforms import validators
+
+
+class IntegerCheck(object):
```

Doesn't WTForms provide these kinds of validators out the box; possibly my not understanding how this integrates with what we already have.

---

Yes, but we can't override their default error, so need to create our own versions of them

---

fair enough

---

Thought about this some more, can we not derive from the WTForms validator and just override/catch/re-throw the exception with our message to re-use as much of the existing validator as possible?

---

The only validator that WTForms supplies that we've created our own version of is the number range - but that's because we want to throw a different error for each limit. There is no Integer validator in wtforms. As we'd need to get right into the call in order to change the message for each limit, this is the simplest solution.

---

Got it 👍